tests/cpp_integration_tests/test_main_model.cpp

```diff
@@ -272,46 +272,6 @@ TEST_CASE("Test main model - power flow") {
     }
 }
 
-TEST_CASE("Test copy main model") { // TODO(mgovers): new test in unit tests: make TestMainModel derived of MainModel,
-                                    // check contents are same after copy (integrity)
-    State state;
-    auto main_model = default_model(state);
-    MainModel model_2{main_model}; // TODO(mgovers): need copy and copy-assign checks in test_api_model.cpp
```

Comment:
I am leaving note because it was removed but the counter part not added yet. I believe in the `test_model_api.cpp` file this can be done. There is a (I believe incomplete) test of copying, so we may just add the remaining cases and check values (this last part is missing I think). That way is verified all the way.

Reply:
cfr. discussions with tony and jerry: we only have default constructors and everything is movable and copyable. this test doesn't add any value

Reply:
`power_grid_model_cpp::Model` does. Unrelated to this one (for now), but we do have custom ones.

Reply:
`power_grid_model_cpp` is a special case because it has explicit constructors and deleters because it's a wrapper of raw C functionality by design. in the core, AFAICT there are no special cases like that